Lib/urllib/parse.py

```diff
@@ -1139,6 +1141,8 @@ def _splitnport(host, defport=-1):
         host = port
     elif port:
         try:
+            if not port.isdigit():
+                raise ValueError(f"Port {port!r} contains non-numeric character(s)")
```

Comment:
Raising a ValueError is odd here because we just catch it immediately below. We can just set `nport` to None directly in this case.

Reply:
Yes, this is a little weird. However, the behavior is documented in the docstring and it's used only from the deprecated `splitnport()` function, so it's probably not worth changing.